### PR TITLE
fix: get the benchmarks run in CI to stick to a constant major ver of node

### PR DESCRIPTION
**Pull request by trentm**
This also bumps that ver from the intended 12 to 14.

Fixes: #2057

### CODE REVIEW

**Comment by trentm** [comment]
jenkins run the benchmark tests

**Comment by trentm** [comment]
Here is a jenkins run that shows the "Benchmarks" stage passing: https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-nodejs%2Fapm-agent-nodejs-mbp/detail/PR-2058/6/pipeline/144

File: .ci/scripts/prepare-benchmarks-env.sh [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
+fi
 
 # This particular configuration is required to be installed in the baremetal
-curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
+curl -sS -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.34.0/install.sh | bash
 export NVM_DIR="$HOME/.nvm"
+set +x  # Disable xtrace because output using nvm.sh is huge.
 [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
 command -v nvm
+nvm --version
```
Comment (astorm) [inline .ci/scripts/prepare-benchmarks-env.sh:23]: just noting that we're outputting the nvm version here -- I presume to aid some future debugging.

**Review by astorm — APPROVED** [review]
✅ approving -- changes seem reasonable.  Including a bit of "here's what I see" for posterity. 

I see we've modified `.ci/Jenkinsfile` to pass in a second argument to the `.ci/scripts/run-benchmarks.sh` program, and that this second argument is the `NODE_VERSION` env var that's set to version 14.

I see we've modified  `.ci/scripts/prepare-benchmarks-env.sh` by

- Adding a `set -x` (and disable this during the `nvm install` run)
- Refusing to run if `NODE_VERSION` is not set
- Making a curl call silent except for errors
- Changing the `nvm install` to always use `NODE_VERSION`

We've modified `.ci/scripts/run-benchmarks.sh` by

- Adding a `set -x` (and disables this during the `nvm install` run)
- Popping its two arguments ($1, $2) into named vars immediately and making them both required (not more default of `apm-agent-benchmark-results.json`)
- Passing `NODE_VERSION` to the `npm run bench:ci` call

We've modified `test/benchmarks/scripts/run-benchmarks-ci.sh` by

- Updating the comments to mention the new version argument
- Adding `"${NODE_VERSION}"` as a new argument to `run-benchmarks.sh`

We've modified `test/benchmarks/scripts/run-benchmarks.sh` by

- Updating the log message with clearer language

**Pull request merged**